src/object_store/src/object/mod.rs

```diff
@@ -508,17 +528,20 @@ impl<OS: ObjectStore> MonitoredObjectStore<OS> {
             .with_label_values(&[self.media_type(), operation_type])
             .start_timer();
 
-        let res = self
-            .inner
-            .read(path, block_loc)
-            .verbose_instrument_await("object_store_read")
-            .await
-            .map_err(|err| {
-                ObjectError::internal(format!(
-                    "read {:?} in block {:?} failed, error: {:?}",
-                    path, block_loc, err
-                ))
-            });
+        let res = tokio::time::timeout(self.read_timeout, async {
```

Comment:
Do we also need a timeout for upload and streaming upload in MonitoredObjectStore?

Reply:
+1 for upload and streaming upload

Reply:
Out of topic, do we need to set speed for write(upload)? Speed is also considered as part of timeout in OpenDAL, for example, if there is no 1024 bytes per second, aka, 1KiB/s, the write will also return TimeoutError.
>For Writer::write(vec![1024*1024*1024])
didn’t return in 60s, it’s ok, we will keep waiting.
didn’t return in 1024s (1GiB/1MiB), we will return a timeout error.